samples/snippets/src/main/java/dialogflow/cx/UpdateIntent.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dialogflow.cx;
+
+import com.google.cloud.dialogflow.cx.v3.Intent;
+import com.google.cloud.dialogflow.cx.v3.Intent.Builder;
+import com.google.cloud.dialogflow.cx.v3.IntentsClient;
+import com.google.cloud.dialogflow.cx.v3.UpdateIntentRequest;
+import com.google.protobuf.FieldMask;
+import java.io.IOException;
+
+public class UpdateIntent {
+
+  public static com.google.cloud.dialogflow.cx.v3.Intent updateIntent(String projectId,
+      String agentId, String intentId, String location, String displayName) throws IOException {
+    IntentsClient client = IntentsClient.create();
+
+    String intentPath =
+        "projects/" + projectId + "/locations/" + location + "/agents/" + agentId + "/intents/"
+            + intentId;
+
+    Builder intentBuilder = client.getIntent(intentPath).toBuilder();
+
+    intentBuilder.setDisplayName(displayName);
+    FieldMask fieldMask = FieldMask.newBuilder().addPaths("display_name").build();
+
+    Intent intent = intentBuilder.build();
+    UpdateIntentRequest request = UpdateIntentRequest.newBuilder()
+        .setIntent(intent)
+        .setLanguageCode("en")
+        .setUpdateMask(fieldMask)
+        .build();
+    return client.updateIntent(request);
```

Comment:
Samples usually return void to show what to with the response https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#method-structure